rules/predicates.py

```diff
@@ -144,11 +144,11 @@ def skip(self):
         """
         raise SkipPredicate()
 
-    def test(self, obj=NO_VALUE, target=NO_VALUE):
+    def test(self, obj=NO_VALUE, target=NO_VALUE, perm=NO_VALUE):
         """
         The canonical method to invoke predicates.
         """
-        args = tuple(arg for arg in (obj, target) if arg is not NO_VALUE)
+        args = tuple(arg for arg in (obj, target, perm) if arg is not NO_VALUE)
```

Comment:
`args` is used as arguments to the predicate, so this means that you're changing the predicate signature `fn(obj, target, rule)` which is undesired. It would be better to add the rule as a new attribute of `Context` so that it will be accessible as:

```
@predicate
def mypred(user, obj):
  rule = mypred.context.rule
  # ...
```